### PR TITLE
Update publish pipeline to avoid repeating git tag step and continue if the error reports that the package is already published

### LINKED ISSUE #7411: Fix publish step in the pipeline
Two current issues in the pipeline:

1. We should only tag the release once, instead of after both scoped and non-scoped package publishes
2. Do a check to see if the package has already been published and, if so, skip the step instead of failing. This will allow us to re-run the step in case there any ephemeral errors in the pipeline that cause the step to fail

### PR DESCRIPTION
**Pull request by sumedhb1995**
- Moves the git tag step up one level so that it only runs once for both scoped and unscoped packages
- Ignore the error on publishing if the error being reported is "You cannot publish over the previously published versions". This allows us to re-run these steps in case there are any failures. This is done by writing the stdout and stderr to a `publish_log` file that we then parse if an error is reported. If the error is for an existing package version, we don't add it our error count

### CODE REVIEW

**Review by tylerbutler — APPROVED** [review]
#️⃣

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 1]
```diff
@@ -88,13 +84,30 @@ steps:
...
       for f in *.tgz
       do
-          if ! npm publish $f $tag ${{ parameters.publishFlags }}
+          npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
+          if grep -q "npm ERR!" "publish_log"
           then
-              let t1+=1
+              if ! grep -q "You cannot publish over the previously published versions" "publish_log"
+              then
+                let t1+=1
```
Comment (curtisman) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:92]: Continue is good for now.  But once we publish topologically, we might want to stop immediately once there is an error.

File: tools/pipelines/templates/include-publish-npm-package-steps.yml [hunk 2]
```diff
@@ -88,13 +84,30 @@ steps:
...
+              else
+                echo Package has already been published.
+                local_f="${f}_local"
+                mv "$f" "$local_f"
+                package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
+                npm pack $package_name
```
Comment (curtisman) [inline tools/pipelines/templates/include-publish-npm-package-steps.yml:98]: Nice cheap way to diff!

**Review by curtisman — APPROVED** [review]
:shipit:

**Pull request merged**